app/models/concerns/procedure_path_concern.rb

```diff
@@ -4,27 +4,66 @@ module ProcedurePathConcern
   extend ActiveSupport::Concern
 
   included do
-    validates :path, presence: true, format: { with: /\A[a-z0-9_\-]{3,200}\z/ }, uniqueness: { scope: [:path, :closed_at, :hidden_at, :unpublished_at], case_sensitive: false }
+    has_many :procedure_paths, inverse_of: :procedure, dependent: :destroy, autosave: true
 
     after_initialize :ensure_path_exists
-    before_save :ensure_path_exists
+    before_validation :ensure_path_exists
+
+    validates :procedure_paths, length: { minimum: 1 }
+
+    scope :find_with_path, -> (path) do
+      normalized_path = path.downcase.strip
+      left_joins(:procedure_paths).where(procedure_paths: { path: normalized_path }).or(where(path: normalized_path)).limit(1)
```

Comment:
```suggestion
      left_joins(:procedure_paths).where(procedure_paths: { path: normalized_path }).or(where(path: normalized_path)).first
```

tout les appelants font un .first, du coup je le rajoute la.